pandas/core/dtypes/dtypes.py

```diff
@@ -305,7 +305,6 @@ def _hash_categories(categories, ordered=True):
                     # everything to a str first, which means we treat
                     # {'1', '2'} the same as {'1', 2}
                     # find a better solution
-                    cat_array = np.array([hash(x) for x in categories])
```

Comment:
iirc this is a check that the categories can be hashed but i think we now check this on construction